### ISSUE TITLE
Add option to inject plugin ID in header for all ascan requests

### ISSUE THREAD

**Issue by zapbot** [zaproxy/zaproxy#1573]
```
Proposed in the User Group: https://groups.google.com/d/msg/zaproxy-users/7lJXZ4hqmFk/Bi8GJ1tfSMgJ

This should be an option, default off.
I think its pretty easy to implement:
In AbstractPlugin.getNewMsg() check for the option, and if turned on then add a header
like: X-ZAP-SCAN-ID with a value of getId()
All ascan rules should use this method when creating new messages to send - if they
dont then they need to change :)

```

Original issue reported on code.google.com by `psiinon` on 2015-03-20 12:39:25

**Comment by thc202** [comment]
Fixed in f5e1bfcf713e242b764b40fab2804b1e5921acec

**Comment by thc202** [comment]
Included in 2.4.1.
